### PR TITLE
Semantic region renderer

### PR DESCRIPTION
**Pull request by Andreas-Forster**
Added a rendering option to render a specific texture image for all faces. This can be used to generate semantic region images. An example is the provided texture where the eyes, the lips, and the ears are colored differently than the skin.

### CODE REVIEW

File: data/config_files/example_config_controlled.json [hunk 1]
```diff
@@ -15,12 +15,17 @@
     "__comment-5" : "render-normals renders visualization of the vertex normals relative to the camera",
     "__comment-6" : "render-albedo renders the albedo of the face instance",
     "__comment-7" : "render-illumination renders a visualization of the illumination",    
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
Comment (BernhardEgger) [inline data/config_files/example_config_controlled.json]: renders a ...

File: data/config_files/example_config_random.json [hunk 2]
```diff
@@ -21,13 +21,15 @@
     "__comment-4" : "render-color-correspondnece-image renders the correspondence to the reference mesh as 3d positions as color coded value",
     "__comment-5" : "render-normals renders visualization of the vertex normals relative to the camera",
     "__comment-6" : "render-albedo renders the albedo of the face instance",
-    "__comment-7" : "render-illumination renders a visualization of the illumination",   
+    "__comment-7" : "render-illumination renders a visualization of the illumination",
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
Comment (BernhardEgger) [inline data/config_files/example_config_random.json]: renders a ...

File: data/config_files/example_config_controlled.json [hunk 3]
```diff
@@ -15,12 +15,17 @@
...
     "render": true,
     "render-depth": true,
     "render-color-correspondence-image": true,
     "render-normals": true,
     "render-albedo": true,
-    "render-illumination": true
+    "render-illumination": true,
+    "render-region-maps": true
+  },
+  "correspondence-maps": {
```
Comment (BernhardEgger) [inline data/config_files/example_config_controlled.json]: is this needed?

File: src/main/scala/faces/utils/TextureMappedPropertyIO.scala [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      "textureMapping" -> textureMappedProperty.textureMapping.toJson,
+      "@type" -> "TextureMappedProperty".toJson
+    )
+
+    val os = new FileOutputStream(new File(directory+"/"+stem+".json"))
+    writeASTToStream(mapping, os)
+  }
+
+//
+//  val refWithTextureMapping = MeshIO.read(new File("/export/faces/model/model2012.2/reference/mean2012_l7_bfm_pascaltex.msh.gz")).get.texturedMesh3D.get
```
Comment (BernhardEgger) [inline src/main/scala/faces/utils/TextureMappedPropertyIO.scala]: remove?

File: src/test/resources/controlledFacesSettings.json [hunk 5]
```diff
@@ -14,13 +14,15 @@
     "__comment-4" : "render-color-correspondnece-image renders the correspondence to the reference mesh as 3d positions as color coded value",
     "__comment-5" : "render-normals renders visualization of the vertex normals relative to the camera",
     "__comment-6" : "render-albedo renders the albedo of the face instance",
-    "__comment-7" : "render-illumination renders a visualization of the illumination",    
+    "__comment-7" : "render-illumination renders a visualization of the illumination",
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
Comment (BernhardEgger) [inline src/test/resources/controlledFacesSettings.json]: renderas a...

File: src/test/resources/randomFacesSettings.json [hunk 6]
```diff
@@ -21,13 +21,15 @@
     "__comment-4" : "render-color-correspondnece-image renders the correspondence to the reference mesh as 3d positions as color coded value",
     "__comment-5" : "render-normals renders visualization of the vertex normals relative to the camera",
     "__comment-6" : "render-albedo renders the albedo of the face instance",
-    "__comment-7" : "render-illumination renders a visualization of the illumination",    
+    "__comment-7" : "render-illumination renders a visualization of the illumination",
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
Comment (BernhardEgger) [inline src/test/resources/randomFacesSettings.json]: renders a ...

**Review by BernhardEgger — COMMENTED** [review]
I have some minor changes.
It would be very cool to get the bfm'09 face segments also as a default mask :)
https://faces.dmi.unibas.ch/bfm/main.php?nav=1-1-0&id=details

Currently the mask is rendered with shadowing and interpolation. For segmentation I think it might be useful to just get a pure color value without illumination effects (as done for other modalities). 
I'm not sure if there is an easy way remove interpolation between two regions

**Comment by BernhardEgger** [comment]
Does the texture mapping work for both face 12 and bfm mask?

**Comment by Andreas-Forster** [comment]
All mentioned issues should be addressed:

- Shading removed from rendering.
- Added specific mappings for BFM and face12 masked models (adapted config files so the mappings match the model).
- Updated comments in config files.
- Removed unused section in a config file.
- Removed out-commented code.

**Comment by Andreas-Forster** [comment]
As an early Christmas present, there is now also the bfm09.png available as region map.

**Comment by BernhardEgger** [comment]
Thank you - now it looks great.
I changed a small typo in the readme.

Happy holidays and I like the present :)

**Review by BernhardEgger — APPROVED** [review]

**Review by AdamKortylewski — APPROVED** [review]
Thanks for the pull request, it's a great feature!

**Pull request merged**